src/WiFi101.h

```diff
@@ -144,10 +146,9 @@ class WiFiClass
 	int hostByName(const char* hostname, IPAddress& result);
 	int hostByName(const String &hostname, IPAddress& result) { return hostByName(hostname.c_str(), result); }
 
-	uint8_t ping(const char* hostname, uint8_t ttl = 128);
-	uint8_t ping(const String &hostname, uint8_t ttl = 128);
-	uint8_t ping(IPAddress host, uint8_t ttl = 128);
-	uint8_t pingGateway();
+	wl_ping_result_t ping(const char* hostname, uint8_t ttl = 128);
```

Comment:
I'd say we go with `uint8_t` for now. Documentation can explain more about the expected return values, like done for [WiFi.status()](https://www.arduino.cc/en/Reference/WiFi101Status).

Eventually the return type of `status()` should probably be `int`. In most/all of the example sketches this is done:

```arduino
// ...
int status = WL_IDLE_STATUS;     // the Wifi radio's status

// ...

void setup() {
  // ... 

  // attempt to connect to Wifi network:
  while ( status != WL_CONNECTED) {
    Serial.print("Attempting to connect to WPA SSID: ");
    Serial.println(ssid);
    // Connect to WPA/WPA2 network:
    status = WiFi.begin(ssid, pass);

    // wait 10 seconds for connection:
    delay(10000);
  }

 // ...
}
```

This seems more "Arduino" style to me.

If we agree on this, the new example sketch should change `byte pingResult;` to `int pingResult;`